### PR TITLE
Added documentation for configuring uPortal with Oracle

### PR DESCRIPTION
**Pull request by jonathanmtran**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added

##### Description of change

Migrated (from https://wiki.jasig.org/display/UPM43/Oracle) and updated documentation for configuring uPortal with Oracle

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by jonathanmtran** [comment]
This is a work in progress. Feedback would be appreciated.

File: docs/database/oracle.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+jdbc "com.oracle:ojdbc:${oracleJdbcVersion}"
+```
+
+`${oracleJdbcVersion}` can be defined `gradle.properties`. Otherwise, substitute `${oracleJdbcVersion}` with the version number used in step 1.
+
+## Step 4: Build and Deploy
+
+You can execute the command below to build the database tables and copy files to your servlet container.
+
+Executing the command `./gradlew portalInit` **will drop and recreate the database tables and all existing data will be lost**. This will result in a clean uPortal database structure. If you want to keep the contents of your existing database, use `./gradlew tomcatDeploy` .
```
Comment (ChristianMurphy) [inline docs/database/oracle.md]: :scroll: just for extra impact I'd add :warning: `:warning:` around the `**will drop and recreate [...] lost**`

File: docs/database/oracle.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Using uPortal with Oracle
+
+## Step 1: Obtain the Driver
+
+Since the Oracle JDBC driver is not available in the central Maven repository, it must be placed into the local repository of each machine on which you wish to build uPortal.
+
+As an alternative to this, you could set up a maven repository for use by multiple machines.
+
+[Download](http://www.oracle.com/technetwork/database/features/jdbc/index-091264.html) the correct Oracle JDBC driver for your server. Once you have the jar, it needs to be installed into the local Maven repository using the following command
```
[the remark this answers is not in the document]
Comment (jonathanmtran) [reply docs/database/oracle.md]: At first glance, my only concern is that the credentials may get inadvertently committed to the repository.

Maybe we opt for the crude method of creating the folder(s) via `mkdir -p $HOME/.m2/repository/com/oracle/ojdbc/${version}/` and dropping the jar in as `ojdbc-${version}.jar`

¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (ChristianMurphy) [reply docs/database/oracle.md]: > the credentials may get inadvertently committed to the repository

A valid concern.
Gradle supports setting credentials from `~/.gradle/gradle.properties` which is a non-*uPortal-start* folder that would not be committed.

> Maybe we opt for the crude method of creating the folder(s) via `mkdir -p $HOME/.m2/repository/com/oracle/ojdbc/${version}/` and dropping the jar in as `ojdbc-${version}.jar`

:man_shrugging:  I'd lean mildly against folder hacking to drop in custom jars.
But the way Oracle has it setup, they don't leave many clean options for integrating the jdbc driver into a package manager.

[the remark this answers is not in the document]
Comment (ChristianMurphy) [reply docs/database/oracle.md]: On further reflection, its probably best to leave as is.

**Pull request merged**